InteractionBase/src/vis/SelectionAtCursorItem.cpp

```diff
@@ -0,0 +1,165 @@
+ ** following conditions are met:
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+#include "SelectionAtCursorItem.h"
+
+#include "VisualizationBase/src/declarative/DeclarativeItemDef.h"
+#include "VisualizationBase/src/declarative/DynamicGridFormElement.h"
+#include "VisualizationBase/src/declarative/GridLayouter.h"
+#include "VisualizationBase/src/items/Text.h"
+#include "VisualizationBase/src/cursor/TextCursor.h"
+#include "QGraphicsColorizeEffect"
+
+namespace Interaction {
+
+ITEM_COMMON_DEFINITIONS(SelectionAtCursorItem, "item")
+
+SelectionAtCursorItem* SelectionAtCursorItem::instance{};
+
+void SelectionAtCursorItem::show(QVector<Model::Node*> selectableNodes, const OnSelectAction onSelect, Item *target,
+								 bool hasTextField, const OnTextAction onText)
+{
+	if (instance)
+		instance->scene()->removeTopLevelItem(instance);
+	instance = new SelectionAtCursorItem(target, selectableNodes, onSelect, hasTextField, onText);
+}
+
+void SelectionAtCursorItem::hide()
+{
+	if (instance)
+		instance->scene()->removeTopLevelItem(instance);
+	instance = nullptr;
+}
+
+SelectionAtCursorItem::SelectionAtCursorItem(Visualization::Item* target, QVector<Model::Node*> selectableNodes,
+		const OnSelectAction onSelect, bool hasTextField, const OnTextAction onText, StyleType* style)
+	: Super(nullptr, style), target_(target), onSelect_(onSelect), onText_(onText), hasTextField_(hasTextField)
+{
+	currentNodes_ = arrange(selectableNodes);
+	target->scene()->addTopLevelItem(this);
```

Comment:
Please do not add an Item to the scene in the item's own constructor. Do this outside.